packages/supermassive/src/codegen/resolvers.ts

```diff
@@ -201,7 +201,10 @@ function createResolversReducer(
     ListType: {
       leave({ type }): ts.TypeNode {
         return createNullableType(
-          factory.createArrayTypeNode(type as ts.TypeNode),
+          factory.createTypeReferenceNode(
+            factory.createIdentifier("ReadonlyArray"),
+            [type as ts.TypeNode],
+          ),
```

Comment:
iirc, the way to do this with modern TS and through the API is something like:

```ts
factory.createTypeOperatorNode(ts.SyntaxKind.ReadonlyKeyword, factory.createArrayTypeNode(type))
```

Which would produce:

```ts
readonly X[]
```